ts/private/ts_project.bzl

```diff
@@ -85,21 +85,21 @@ See https://github.com/aspect-build/rules_ts/issues/228 for more details.
     # Add user specified arguments *before* rule supplied arguments
     arguments.add_all(ctx.attr.args)
 
-    outdir = _lib.join(ctx.label.package, ctx.attr.out_dir) if ctx.attr.out_dir else ctx.label.package
-    if outdir == "":
-        outdir = "."
+    outdir = _lib.join(
+        ctx.label.workspace_root,
+        _lib.join(ctx.label.package, ctx.attr.out_dir) if ctx.attr.out_dir else ctx.label.package,
+    )
+    tsconfig_path = ctx.file.tsconfig.path.removeprefix("%s/" % ctx.bin_dir.path)
     arguments.add_all([
         "--project",
-        ctx.file.tsconfig.short_path,
+        tsconfig_path,
```

Comment:
I think this can be simplified to,

```
to_output_relative_path(ctx.file.tsconfig)
```

and it will be correct even in the unexpected case where the file is not an output file.

you can load that from

```
load("@aspect_bazel_lib//lib:paths.bzl", "to_output_relative_path")
```

Reply:
Done. This got slightly more complicated as it turns out `ctx.file.tsconfig.path` is misleading because it referenced the raw source file _before_ it was copied to bin, so `to_output_relative_path()` was using a bad input. Updated the rule to copy to bin and the very start with the rest of the sources so everything would use the copied file.